### ISSUE TITLE
S4670: Selectors should be known vs Custom Elements

### ISSUE THREAD

**Issue by miqmago** [SonarSource/sonar-css#358]
With the use of frameworks like angular or in web components, where you can define your own web components, this rule reports as bug something it's actually not. Shouldn't it be reported as info instead of bug?

I.e., when using [camera-component](https://www.npmjs.com/package/camera-component) something like this will be reported as bug:

```css
camera-component {
    padding: 2px;
}
```

As stated in https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-core-concepts, custom elements:

> A valid custom element name is a sequence of characters name that meets all of the following requirements:
> 
> - name must match the PotentialCustomElementName production:
> 
> PotentialCustomElementName ::=
> [a-z] (PCENChar)* '-' (PCENChar)*
>
> PCENChar ::=
> "-" | "." | [0-9] | "_" | [a-z] | #xB7 | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x203F-#x2040] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
> This uses the EBNF notation from the XML specification. [XML]
> 
> - name must not be any of the following:
> 
> annotation-xml
> color-profile
> font-face
> font-face-src
> font-face-uri
> font-face-format
> font-face-name
> missing-glyph

**Comment by chris-serra** [comment]
See this issue: #265 

Our team added the Stylelint rule to our .stylelintrc file, and Sonar-CSS now ignores the violations.

But I see you're looking to add this as an automatic exemption by Sonar-CSS based on the definition of a custom element.

**Comment by miqmago** [comment]
Ouch! Sorry I'm not an expert, we are using sonar-scanner with drone:

```yml
kind: pipeline
type: docker
name: default

steps:
  - name: install
    image: node:12-alpine
    commands:
      # Install node_modules
      - npm i

  - name: sonar-scanner
    image: aosapps/drone-sonar-plugin
    commands:
      - sonar-scanner
```

and in sonar-project.properties:
```properties
sonar.host.url=https://sonar-server.com
sonar.projectKey=awesome-app
sonar.projectName=awesome-app
sonar.projectVersion=0.1.0
sonar.login=9f...
sonar.sources=./src
sonar.exclusions=coverage/**, logs/**, node_modules/**, scripts/**, test/**, typings/**
sonar.javascript.lcov.reportPaths=coverage/lcov.info
```

What I see in our https://sonar-server.com Community Edition Version 8.9 (build 43852) is that all these css properties raise a bug. I'm not sure how to set up the `ignoreTypes` or any other configuration, maybe on https://sonar-server.com or in command line.

What we have done up to now is to duplicate the css rules and deactivate the S4670 rule, but I would rather prefer to have it as an info, just to force reviewing it at any time. Don't know if it is possible...

**Comment by chris-serra** [comment]
You would [add Stylelint to your project](https://stylelint.io/user-guide/get-started/)

And add something like this to the `rules` block in your `.stylelint.json`:

```
  "rules": {
   ...
    "selector-type-no-unknown": [ true, {
        "ignoreTypes": ["/^[a-zA-Z]([a-zA-Z0-9]*-[a-zA-Z0-9]+)+/"]
    } ],
   ...
  }
```

Here's the rule to see how else you can configure it: https://github.com/stylelint/stylelint/tree/master/lib/rules/selector-type-no-unknown

**Comment by vilchik-elena** [comment]
@miqmago Are you sure you are using latest version? There was a fix in March to avoid those FPs

**Comment by yassin-kammoun-sonarsource** [comment]
Closing as no response from the user and a fix was applied to the rule.
